### PR TITLE
Retain old data in Muon GUI

### LINKED ISSUE #24676: Update new muon GUI so not delete old data when new data loaded.
Currently when a new run is loaded the settings in the GUI are reset. The settings should in fact be maintained from the previous run if such settings exist and the settings from old runs should be maintained to be reapplied if that run is switched back to.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This PR updates the GUI to retain old data once new data is loaded and not reset everything whenever new data is loaded or the instrument is changed. 

There are also changes made so that the run selector in the load widget can more easily be used as a selector for which runs to process later on. To aid with this checks are added so data is not reloaded if not necessary and so that data which fails to load is not selected.

In addition to this the load options are changed so that loading a single run is treated as a special case of loading multiple runs.

**Report to:** anthony 

**To test:**

 * Code review
 * Play around with loading and data loading new data and changing instrument.

Fixes #24676  

*This does not require release notes* because the GUI is not yet user faceing

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/utilities/muon_load_data_test.py [hunk 1]
```diff
@@ -187,15 +194,15 @@ def test_that_can_add_data_to_struct(self):
         self.assertEqual(self.muon_load_data.get_parameter('workspace')[1], self.workspace)
         self.assertEqual(self.muon_load_data.get_parameter('filename')[0], 'path to file')
 
-    def test_that_matches_returns_true_for_all_entries_with_one_match(self):
+    def test_that_matches_returns_false_for_all_entries_with_only_one_match(self):
         match_list = self.muon_load_data._matches(run=1, workspace=self.workspace, filename='matching path')
 
-        self.assertEqual(match_list, [True, True, True, False, False])
+        self.assertEqual(match_list, [False, False, False, False, False])
```
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: why have these changed

File: scripts/test/Muon/utilities/muon_load_data_test.py [hunk 2]
```diff
@@ -187,15 +194,15 @@ def test_that_can_add_data_to_struct(self):
...
 
-    def test_that_matches_with_no_params_matches_none(self):
+    def test_that_matches_with_no_params_matches_all(self):
         match_list = self.muon_load_data._matches()
 
-        self.assertEqual(match_list, [False, False, False, False, False])
+        self.assertEqual(match_list, [True, True, True, True, True])
```
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_load_data_test.py]: same as above

**Comment by AnthonyLim23** [comment]
If I click the back/forward buttons for the runs it automatically has both in the box. Is that intentional? 

Otherwise I think its ok

**Comment by AnthonyLim23** [comment]
found a bug.

Load some data for EMU
Click back
Change instrument to HIFI
load some data
Change instrument to MUSR
load some data
press back
The run button will have all of the runs you have loaded (even from the wrong instrument)

**Comment by AnthonyLim23** [comment]
Had a thought about this at the weekend. What are we doing about the calculate alpha button when multiple runs are loaded?

**Comment by AnthonyLim23** [comment]
If I make a plot from the data and then load a new run it gets wiped :(

**Comment by AnthonyLim23** [comment]
If I load a run in the new GUI and the old one then the group and pairs have the wrong x values (even when both GUI's have use values from file ticked)

**Comment by AnthonyLim23** [comment]
it is looking good. Found another bug. If you load HIFI 84447 (multi period) and change the period to 2, then click update all an error is thrown

**Comment by AnthonyLim23** [comment]
minor point, if I load multiple runs then the new alpha pop up works well. If I then turn on multiple runs and press guess alpha I still get the pop up (I would expect to only be looking at a single run)

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by martyngigg** [comment]
Failing tests are unrelated.

**Pull request merged**